### DISCUSSION TITLE
How can I emulate the behavior of Spring's "InitializingBean" on Quarkus ?

### DISCUSSION THREAD

**Discussion by p4535992** [quarkusio/quarkus#39117] [Q&A] [answered]
I am migrating some spring projects to quarkus , what is the best way to emulate the old spring "InitializingBean" interface on quarkus ?

[a comment by an automated account is not shown]

**Reply by geoand** [reply, marked as answer]
Have a look at https://quarkus.io/guides/lifecycle#listening-for-startup-and-shutdown-events

**Reply by p4535992** [reply]
Ty

**Comment by p4535992** [comment]
For anyone with the same doubt
 ```
 @Startup
   void init() { 
     doSomeCoolInit();
   }
```
